src/java/org/apache/cassandra/repair/StreamingRepairTask.java

```diff
@@ -79,7 +79,7 @@ public void run()
     @VisibleForTesting
     StreamPlan createStreamPlan(InetAddressAndPort dest)
     {
-        StreamPlan sp = new StreamPlan(StreamOperation.REPAIR, 1, false, pendingRepair, previewKind)
+        StreamPlan sp = new StreamPlan(desc.determanisticId(), StreamOperation.REPAIR, 1, false, pendingRepair, previewKind)
```

Comment:
the idea is that streaming done by repair will use the RepairJob id (as defined by CASSANDRA-15399), so if a repair exists with the same planId, then this stream can map to the repair without any networking changes

Reply:
ok, so this may not work as well as I wanted, and seems we don't actually test multiple streams well!

in repair we get back N trees, and we check all pairs to see if there is a mismatch (coordinator may match peer1, but peer1 and peer2 may not match, so we ask them to fix themselves); so we can have multiple streams from a single RepairJob.  Now, for the coordinator to know what streams are used, it would make sense to have coordinator control the ID, but this trick won't work.

I will revert these changes as they don't work as planned, need to also improve testing to trigger this case.